docs/choreolib/auto-factory.md

```diff
@@ -1,109 +1,212 @@
-    autoChooser.addRoutine("twoPieceAuto", this::twoPieceAuto);
-  }
-
-  // this would normally be in a separate file
-  private AutoRoutine twoPieceAuto(AutoFactory factory) {
-    final AutoRoutine routine = factory.newRoutine("twoPieceAuto");
-
-    final AutoTrajectory trajectory = routine.trajectory("twoPieceAuto");
-
-    routine.running()
-        .onTrue(
-            drive.resetOdometry(
-                    trajectory.getInitialPose()
-                        .orElseGet(
-                            () -> {
-                              routine.kill();
-                              return new Pose2d();
-                            }))
-                .andThen(trajectory.cmd())
-                .withName("twoPieceAuto entry point"));
-
-    trajectory.atTime("intake").onTrue(intake.extend());
-    trajectory.atTime("shoot").onTrue(shooter.launch());
+Trigger myTrigger = new Trigger(() -> condition);
+
+// Safe
+routine.observe(myTrigger).onTrue(Commands.print("Foo"));
+routine.running().and(myTrigger).onTrue(Commands.print("Bar"));
+
+// Unsafe
+myTrigger.onTrue(Commands.print("Foo"));
+myTrigger.and(routine.running()).onTrue(Commands.print("Bar"));
+```
+
+Trajectories can be loaded using `AutoRoutine.trajectory()`, which will return an `AutoTrajectory` ([Java](/api/choreolib/java/choreo/auto/AutoTrajectory.html)). The `AutoTrajectory` class exposes multiple triggers for you to attach reactive logic to, as well as `AutoTrajectory.cmd()` for scheduling the trajectory.
+
+```java
+public AutoRoutine pickupAndScoreAuto() {
+    AutoRoutine routine = autoFactory.newRoutine("pickupAndScore");
+
+    // Load the routine's trajectories
+    AutoTrajectory pickupTraj = routine.trajectory("pickupGamepiece");
+    AutoTrajectory scoreTraj = routine.trajectory("scoreGamepiece");
+
+    // When the routine begins, start the first trajectory
+    routine.running().onTrue(pickupTraj.cmd()); // TODO Reset Odometry (Pending API change)
+
+    // Starting at the event marker named "intake", run the intake
+    pickupTraj.atPose("intake").onTrue(intakeSubsystem.intake());
+
+    // When the trajectory is done, start the next trajectory
+    pickupTraj.done().onTrue(scoreTraj.cmd());
+
+    // While the trajectory is active, prepare the scoring subsystem
+    scoreTraj.active().whileTrue(scoringSubsystem.getReady());
+
+    // When the trajectory is done, score
+    scoreTraj.done().onTrue(scoringSubsystem.score());
 
     return routine;
-  }
+}
+```
+
+Sometimes, you may want to implement a "branching auto": an autonomous routine that changes behavior based on the state of the robot. An excellent example for the need of branching autos is the [2024 season](https://youtu.be/9keeDyFxzY4), where robots would race to the midline to grab a gamepiece. If another robot beat yours to the midline, or your robot missed a game piece, a common strategy was to go directly to the next gamepiece on the midline, instead of coming back to score.
```

Comment:
A video of the actual autos your describing would be better here

Reply:
Funnily enough when I originally wrote this section I spent ~20 minutes trying to find an example from Chezy Champs (for better video quality) but unfortunately I couldn't find any good examples, I'll take a look at MTTD to see if I can find anything there, if you have a match in mind let me know and I'll add it in, this was another task I forgot to add to the to do list

Reply:
I don't have a match but I remember a YouTube short which probably describes it well enough if you can't find a match